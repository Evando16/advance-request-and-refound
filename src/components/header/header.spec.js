import React from 'react';
import { render } from '@testing-library/react';
import pretty from 'pretty';
import Header from './header';

describe('HeaderComponent', () => {
  const headerDataMock = {
    accountabilityExtraInfo: {
      amountOfPeople: 1,
      budgetForBreakfast: 'NÃ£o',
      eventDate: '05/14/2020',
    },
    collaborator: {
      email: 'quickops-bs@hotmart.com',
      id: 10436247,
      name: 'Quickops BS',
    },
    costCenters: [
      {
        id: 88,
        name: '01 - Sem manager',
      },
    ],
    id: 756,
    justification: 'f',
    project: {
      id: 41,
      title: 'Afiliados Brasil',
    },
    purpose: 'Fraternization',
    type: 'Refund',
  };

  let container = null;

  beforeEach(() => {
    container = render(<Header headerData={headerDataMock} />);
  });

  describe('Snapshot tests', () => {
    it('should render header component', async () => {
      expect(pretty(container.innerHTML)).toMatchSnapshot();
    });

    it('should render header component with conditional fields', async () => {
      container = render(<Header headerData={{ ...headerDataMock, project: null }} />);

      expect(pretty(container.innerHTML)).toMatchSnapshot();
    });
  });
});
