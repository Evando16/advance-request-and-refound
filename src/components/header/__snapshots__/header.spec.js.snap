// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent Snapshot tests should render header component 1`] = `""`;

exports[`HeaderComponent Snapshot tests should render header component with conditional fields 1`] = `""`;
